askametric/query_processor/query_processor.py

```diff
@@ -426,12 +454,17 @@ async def process_query(self) -> None:
 
         if self.guardrails.relevant is False:
             self.final_answer = self.guardrails.relevance_response
-            await self._translate_final_answer()
+            await self._get_translated_final_answer()
             return None
 
-        # Step through rest of pipeline
-        await self._get_best_tables_from_llm()
-        await self._get_best_columns_from_llm()
-        await self._get_sql_query_from_llm()
-        await self._get_final_answer_from_llm()
-        await self._translate_final_answer()
+        if (self.query_type == 1) or (self.query_type == 2):
```

Comment:
Yes, I didn't want to do it because I didn't want to refactor LLMQueryProcessor rn. We can discuss the separation that you suggested at tech session / sprint planning, and happy to take a ticket to refactor in the future.